web/new_alt_text_manager.js

```diff
@@ -508,40 +514,69 @@ class ImageAltTextSettings {
       this.#togglePref.bind(this, "enableNewAltTextWhenAddingImage")
     );
 
-    deleteModelButton.addEventListener("click", async () => {
-      await mlManager.deleteModel("altText");
+    deleteModelButton.addEventListener("click", this.#delete.bind(this, true));
+    downloadModelButton.addEventListener(
+      "click",
+      this.#download.bind(this, true)
+    );
 
-      aiModelSettings.classList.toggle("download", true);
-      createModelButton.disabled = true;
-      createModelButton.setAttribute("aria-pressed", false);
-      this.#setPref("enableGuessAltText", false);
-      this.#setPref("enableAltTextModelDownload", false);
+    closeButton.addEventListener("click", this.#finish.bind(this));
+
+    eventBus._on("enablealttextmodeldownload", ({ value }) => {
+      if (value) {
+        this.#download(false);
+      } else {
+        this.#delete(false);
+      }
     });
 
-    downloadModelButton.addEventListener("click", async () => {
-      downloadModelButton.disabled = true;
-      downloadModelButton.firstChild.setAttribute(
-        "data-l10n-id",
-        "pdfjs-editor-alt-text-settings-downloading-model-button"
-      );
+    this.#overlayManager.register(dialog);
+  }
 
-      await mlManager.downloadModel("altText");
+  async #download(isFromUI) {
+    this.#downloadModelButton.disabled = true;
+    this.#downloadModelButton.firstChild.setAttribute(
+      "data-l10n-id",
+      "pdfjs-editor-alt-text-settings-downloading-model-button"
+    );
 
-      aiModelSettings.classList.toggle("download", false);
-      downloadModelButton.firstChild.setAttribute(
-        "data-l10n-id",
-        "pdfjs-editor-alt-text-settings-download-model-button"
-      );
-      createModelButton.disabled = false;
-      createModelButton.setAttribute("aria-pressed", true);
+    if (isFromUI) {
+      await this.#mlManager.downloadModel("altText");
+    }
+
+    this.#aiModelSettings.classList.toggle("download", false);
+    this.#downloadModelButton.firstChild.setAttribute(
+      "data-l10n-id",
+      "pdfjs-editor-alt-text-settings-download-model-button"
+    );
+
+    this.#createModelButton.disabled = false;
+    this.#createModelButton.setAttribute("aria-pressed", true);
+    if (!isFromUI) {
       this.#setPref("enableGuessAltText", true);
-      mlManager.toggleService("altText", true);
+    }
+
+    this.#mlManager.toggleService("altText", true);
+    if (isFromUI) {
       this.#setPref("enableAltTextModelDownload", true);
-      downloadModelButton.disabled = false;
-    });
+    }
 
-    closeButton.addEventListener("click", this.#finish.bind(this));
-    this.#overlayManager.register(dialog);
+    this.#downloadModelButton.disabled = false;
```

Comment:
Won't this just override the value [set above](https://github.com/mozilla/pdf.js/pull/18573/files#diff-6dbe68273ba0858959b04fe0ca8097d9cf9c8a9fccc75dfc8ef222fbcf12ac07R537), meaning that either of these lines seem redundant?